### PR TITLE
Add Counters with an in-memory backend.

### PR DESCRIPTION
**Pull request by mythmon**
I'm pretty sure my model changes are going to conflict with #3. I can rework the model parts of this after that lands.

@Osmose can you r? with the caveat that I'm going to re-do the model changes?

### CODE REVIEW

File: normandy/counters/__init__.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+class CounterBackend:
```
Comment (Osmose) [inline normandy/counters/__init__.py]: Newstyle classes 4 lyfe

File: normandy/counters/__init__.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+class Counter:
+
+    def __init__(self):
+        self.backend = _counter_backend()
+
+    def _recipe_name(self, recipe):
+        return 'recipe-{}'.format(recipe.pk)
+
+    def check(self, recipe):
```
Comment (Osmose) [inline normandy/counters/__init__.py]: `needs_more_hits`? `can_deliver`?

File: normandy/counters/__init__.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        count = self.backend.get(name)
+        return count < recipe.count_limit
+
+    def increment(self, recipe):
+        if not recipe.has_counter:
+            return True
+
+        name = self._recipe_name(recipe)
+        self.backend.increment(name)
+
```
Comment (Osmose) [inline normandy/counters/__init__.py]: Return False? And also maybe a docstring describing what the return value means.

File: normandy/counters/__init__.py (hunk 1 above)
Comment (mythmon) [inline normandy/counters/__init__.py]: Sorry, I forgot. Should I add `(object)` to this? Or just leave it?

**Comment by Osmose** [comment]
r+

**Comment by mythmon** [comment]
^ feedbacked

**Comment by Osmose** [comment]
r++

**Pull request merged**